src/helper_functions.jl

```diff
@@ -4,8 +4,8 @@ export _At_mul_B
 const DEFAULT_COND_TOL = 1e6
 
 # matrix-matrix multiplication and division
-@inline _At_mul_B(A, B) = transpose(A) * B
-@inline _At_ldiv_B(A, B) = transpose(A) \ B
+@inline _At_mul_B(A, B) = transpose(Matrix(A)) * B
+@inline _At_ldiv_B(A, B) = transpose(Matrix(A)) \ B
```

Comment:
why?

Reply:
I believe it was an old workaround for using projection (and Linear Map) for `Halfspaces` on specific variables. We discussed it at some point with @mforets. Maybe now he has some better ideas on how to fix it.

Reply:
`transpose` works for sparse matrices. Do you have a use case?

Reply:
We can transpose standard `SparceArray` with type `SparseMatrixCSC{Float64,Int64}`:
```
julia> sp = spzeros(1,4)
1×4 SparseMatrixCSC{Float64,Int64} with 0 stored entries

julia> sp[1,1] = 1
1

julia> transpose(sp)
4×1 LinearAlgebra.Transpose{Float64,SparseMatrixCSC{Float64,Int64}}:
 1.0
 0.0
 0.0
 0.0
```
However, inside helper function we somehow get sparce matrix from `LinearAlgebra` package with the type `SuiteSparse.SPQR.QRSparse{Float64,Int64}` 
```
julia> lspar = qr(sp)
SuiteSparse.SPQR.QRSparse{Float64,Int64}
Q factor:
1×1 SuiteSparse.SPQR.QRSparseQ{Float64,Int64}:
 1.0
R factor:
1×4 SparseMatrixCSC{Float64,Int64} with 1 stored entry:
  [1, 1]  =  1.0
Row permutation:
1-element Array{Int64,1}:
 1
Column permutation:
4-element Array{Int64,1}:
 1
 2
 3
 4

julia> transpose(lspar)
ERROR: MethodError: no method matching transpose(::SuiteSparse.SPQR.QRSparse{Float64,Int64})
```

Reply:
Then the Julia paradigm is to define new specific methods for these cases instead of making all other cases suffer:
```julia
@inline _At_mul_B(A::Factorization, B) = transpose(Matrix(A)) * B
@inline _At_ldiv_B(A::Factorization, B) = transpose(Matrix(A)) \ B
```
I found the type for `A` like this:
```julia
julia> supertype(typeof(lspar))
Factorization{Float64}
```

Reply:
Actually now I am confused. It seems that we actually have standard sparce matrix, but the error says opposite:
```
julia> d_int_g = intersection(cpa1, G)
HPolytope{Float64}
LinearMap{Float64,HPolyhedron{Float64},Float64,SparseMatrixCSC{Float64,Int64}}
Type of M is.. SparseMatrixCSC{Float64,Int64}

ERROR: MethodError: no method matching transpose(::SuiteSparse.SPQR.QRSparse{Float64,Int64})
``` 
(I have added my own output to debug...)

Reply:
Fot testing purposes you can use:

```
i1 = Interval(N[0, 1])
i2 = Interval(N[2, 3])
h1 = Hyperrectangle(low=N[3, 4], high=N[5, 7])
cpa1 = CartesianProductArray([i1, i2, h1])
G = HPolyhedron([HalfSpace(N[1, 0, 0, 0], N(1))])
d_int_g = intersection(cpa1, G)
```

Reply:
My only guess is that you do not `transpose` this matrix directly. Best look at the line where it occurs.

Reply:
So as I can see in linalg.jl, since matrix M is not squared it calls `LinearAlgebra`'s sparce matrix:
https://github.com/JuliaLang/julia/blob/master/stdlib/SparseArrays/src/linalg.jl#L1400